tests/linalg_test.py

```diff
@@ -783,6 +783,37 @@ def testMultiDot(self, shapes, dtype, rng_factory):
     self._CompileAndCheck(jnp_fun, args_maker, check_dtypes=True,
                           atol=tol, rtol=tol)
 
+  @parameterized.named_parameters(jtu.cases_from_list(
+      {"testcase_name":
+       "_lhs={}_rhs={}_rcond={}".format(
+           jtu.format_shape_dtype_string(lhs_shape, dtype),
+           jtu.format_shape_dtype_string(rhs_shape, dtype),
+           rcond),
+       "lhs_shape": lhs_shape, "rhs_shape": rhs_shape, "dtype": dtype,
+       "rcond": rcond, "rng_factory": rng_factory}
+      for lhs_shape, rhs_shape in [
+          ((1, 1), (1, 1)),
+          ((4, 6), (4,)),
+          ((6, 6), (6, 1)),
+          ((8, 6), (8, 4)),
+      ]
+      for rcond in [-1, None, 0.5]
+      for dtype in float_types + complex_types
+      for rng_factory in [jtu.rand_default]))
+  @jtu.skip_on_devices("tpu")  # SVD not implemented on TPU.
+  def testLstsq(self, lhs_shape, rhs_shape, dtype, rcond, rng_factory):
+    rng = rng_factory()
+    _skip_if_unsupported_type(dtype)
+    onp_fun = partial(onp.linalg.lstsq, rcond=rcond)
+    jnp_fun = partial(np.linalg.lstsq, rcond=rcond)
+    jnp_fun_numpy = partial(np.linalg.lstsq, rcond=rcond, numpy_resid=True)
+    args_maker = lambda: [rng(lhs_shape, dtype), rng(rhs_shape, dtype)]
+    tol = {onp.float32: 1e-6, onp.float64: 1e-12,
+           onp.complex64: 1e-6, onp.complex128: 1e-12}
+
+    self._CheckAgainstNumpy(onp_fun, jnp_fun_numpy, args_maker, check_dtypes=False, tol=tol)
+    self._CompileAndCheck(jnp_fun, args_maker, check_dtypes=True, atol=tol, rtol=tol)
+
```

Comment:
I think it would be a good idea to add a `check_grads()` call here to verify gradients, given how fragile gradients are for some of the operations involved here (like SVD or potential division by zero)

Reply:
I gave this a try... it fails with a dtype casting error in test_util. Perhaps due to the fact that some of the outputs are integer type?

I commented-out the test for now.

Reply:
Yes, that makes sense. I would suggest just testing the first output from `lstsq` then, via a wrapper function.

Reply:
e.g., `jtu.check_grads(lambda *args: jnp_fun(*args)[0], args_maker(), order=2, atol=tol, rtol=tol)`

Reply:
Thanks! That works - 32-bit was having issues, so I limited it to 64-bit similar to the check in Cholesky.